File: Systems-for-Design-and-Implementation/Exam/Schita/ExamenMPP/ExamenMPP/Networking/src/main/java/app/network/jsonprotocol/Request.java
package app.network.jsonprotocol;

// TODO 1: IMPORT MODELS FOR REQUEST CLASS

public class Request {

    private RequestType type;
    private String stringR;
    private Long aLong;

    // TODO 2: ADD MODELS FOR REQUEST CLASS

    public Request(){}

    public RequestType getType() {
        return type;
    }

    public void setType(RequestType type) {
        this.type = type;
    }

    public String getStringR() {
        return stringR;
    }

    public void setStringR(String stringR) {
        this.stringR = stringR;
    }

    public Long getLong() {
        return aLong;
    }

    public void setLong(Long aLong) {
        this.aLong = aLong;
    }

    /* TODO 3: CREATE toString() FOR REQUEST CLASS
    @Override
    public String toString() {
        return "Request{" +
                "type=" + type +
                ", reservationManager=" + reservationManager +
                ", trip=" + trip +
                ", reservation=" + reservation +
                ", seat=" + seat +
                '}';
    }
     */
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab11/ProiectCSharp/TransportNetworking/transport.network/jsonprotocol/Request.cs
using System.Collections.Generic;
using Transport.Model.transport.model;
using TransportModel.transport.model;

namespace TransportNetworking.transport.network.jsonprotocol
{
    public class Request
    {
        public RequestType type { get; set; }
        public string stringR { get; set; }
        public long? aLong { get; set; }
        public ReservationManager reservationManager { get; set; }

        public Trip trip { get; set; }

        public Seat seat { get; set; }
        public Reservation reservation { get; set; }
        public Trip[] trips { get; set; }
        public List<Reservation> reservations { get; set; }
        public List<Seat> seats { get; set; }

        public override string ToString()
        {
            return $"Request{{type={type}, reservationManager={reservationManager}, trip={trip}, reservation={reservation}, seat={seat}}}";
        }
    }
}